requests-for-comments/2025-09-24-10x-engineering-onboarding.md

```diff
@@ -0,0 +1,88 @@
+# Request for comments: 10x-ing PostHog's engineering onboarding
+
+**Decision maker:** Engineering team leads + the blitzscale folks
+
+## Overview
+
+Our engineering onboarding is inconsistent and doesn't scale. Teams handle it in isolation, creating wildly different experiences that make it impossible to assess performance or teach our ways of working effectively.
+
+This RFC proposes cohort-based onboarding (HogBoarding) that standardizes the experience while building stronger cross-team connections.
+
+## Why now
+
+1. **We're hiring rapidly** and inconsistent onboarding amplifies all the problems
+2. **Recent onboarding failures** - several new hires haven't ramped effectively, leading to longer time-to-productivity and people leaving during probation
+3. **Internal poaching is breaking down** - we've moved people from overstaffed teams (Feature Flags) without communicating expectations upfront, creating confusion about hiring plans
+
+## Problem
+
+Current onboarding has too much variance in quality, operates in silos, doesn't teach the PostHog way effectively, creates an "onboarding tax" on high-performing teams, and makes inefficient use of our technical debt.
+
+**More details:**
+
+1. **Variance in quality.** Some teams have well-thought-out processes, others wing it. New hires get wildly different experiences depending on which team they join, making it impossible to assess performance consistently or ensure everyone learns our core ways of working.
+
+2. **Teams operate in silos.** New engineers only meet their immediate team members during their first weeks. This limits cross-team collaboration and means people miss out on learning from other parts of the product and codebase.
+
+3. **Not teaching the PostHog way effectively.** Our core values around autonomy, ownership, and development practices are best learned through experience, but we're trying to teach them within the narrow context of a single product area.
+
+4. **"Onboarding tax" on high-performing teams.** Teams with strong onboarding culture (e.g. Feature Flags) end up overhiring, expecting that ramped-up engineers will get poached by teams that need them. This creates weird expectations—how do teams know they're supposed to overhire? It also unfairly burdens teams that invested in good onboarding processes.
+
+5. **Inefficient use of starter tasks.** We have technical debt and starter tasks across all products, but currently these only get tackled by people on those specific teams. We're missing an opportunity to pay down this debt while giving new hires diverse learning experiences.
```

Comment:
I feel like we never actually have that many starter tasks (as "debt") because lots of simple things just get done. Though I do think we could have better hygiene around creating github issues for things we aren't going to just do and labeling them as good-first-issue so they are easier to find.